File: app/src/main/java/sk/company/androidboilerplate/injection/component/ActivityComponent.java
package sk.company.androidboilerplate.injection.component;

import dagger.Subcomponent;
import sk.company.androidboilerplate.injection.PerActivity;
import sk.company.androidboilerplate.injection.module.ActivityModule;
import sk.company.androidboilerplate.ui.main.MainActivity;

/**
 * This component inject dependencies to all Activities across the application
 */
@PerActivity
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);

}

File: app/src/main/kotlin/sk/company/androidboilerplate/injection/component/ActivityComponent.kt
package sk.company.androidboilerplate.injection.component

import dagger.Subcomponent
import sk.company.androidboilerplate.injection.PerActivity
import sk.company.androidboilerplate.injection.module.ActivityModule
import sk.company.androidboilerplate.ui.main.MainActivity

@PerActivity
@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun inject(kotlinActivity: MainActivity)
}